### DISCUSSION TITLE
MODBUS Problem

### DISCUSSION THREAD

**Discussion by CPESSEN** [artisan-roaster-scope/artisan#694] [Q&A]
Hey Experts, 

thx for having me here.

Now I got an issue while roasting with artisan and probation 12.

since few days my Artisan crashes the I start an new roast. I set the Heattemp and start the roast. a few moments later the whole application crashes an closes itself with an immediate reboot. Within the logs I get the message "MODBUS Connection lost" and after reboot I get the message "Modbis connection reestablished"

Im Running the software Version 2.4.6 Built 2 on an iMac with latest BisSur. and wired network

I checked the Networking Situation as well but there are no connection losses oder Problems within the log files.

Dows anyone have an idea?

thanks in advance

**Comment by MAKOMO** [comment]
> Hey Experts,
> 
> thx for having me here.
> 
> Now I got an issue while roasting with artisan and probation 12.

Which exact model and build year?

> since few days my Artisan crashes the I start an new roast. 

What did change a few days ago? I assume Artisan was running stable before, right?

What exactly do you mean by "start a new roast"? The moment you press ON or START or the moment you press CHARGE?

> I set the Heattemp and start the roast. a few moments later the whole application crashes an closes itself with an immediate reboot. 

Hm. the app crashes. But what does reboot? Is it the computer?

> Within the logs I get the message "MODBUS Connection lost" and after reboot I get the message "Modbis connection reestablished"

Reboot of the computer or restart of the app?

> 
> Im Running the software Version 2.4.6 Built 2 on an iMac with latest BisSur. and wired network
> 
> I checked the Networking Situation as well but there are no connection losses oder Problems within the log files.
> 
> Dows anyone have an idea?
> 
> thanks in advance

Could you please attach to this discussion here on Github your artisan-settings.txt file by a simple drag-and-drop to the message field (export the Artisan settings via menu Help >> Save Settings and rename artisan-settings.aset to artisan-settings.txt). I will have a look.

**Reply by CPESSEN** [reply]
Hey Makomo,

Model Probaten Typ 2 SSG2 PC Year 2016

Nothing really changed. Maybe one of my employees had accidentally updated the Artisan Software, but none of them can "remember"... Sorry

By start a new roast I mean that I open the software, load my roast image, click on start an set the temperature of the burner. the it ist ready I open the valve to fill in the raw beans. and exactly then the applications crashes an close (it is just gone). but the next second it restarts with an new profile (roast clock starts again on 0sec).

The computer does not reboot. only the application.
I have attached the file.

thanks in Advance
[artisan-einstellungen.txt](https://github.com/artisan-roaster-scope/artisan/files/7197096/artisan-einstellungen.txt)

**Comment by MAKOMO** [comment]
I am still scratching my head on this one. Your settings look absolute fine to me. The crash must happen on Artisan detecting your bean filling action (you have autoCHARGE ticked). 

Some further more questions:
- does the crash happen always and on each roast?
- does the app opens the Roast Properties dialog before crashing (you have the "Open on CHARGE" flag ticked)?
- Artisan does not has a mechanism to automatically restart on crashes. I wonder what is triggering this restart. Is maybe your system asking via a dialog ("app has crashed. Restart?" or similar) to restart the crashed app?

In case you are located in Germany, can we have a quick chat in German on the phone? If you are available, please send me your telephone number by email to info@artisan.plus and I will call you.

**Comment by MAKOMO** [comment]
Any update on this one?
